### PR TITLE
New Resource: `azurerm_disk_pool`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR introduces a new resource `azurerm_disk_pool` which supersedes the existing `azurerm_storage_disks_pool` - this is because the existing resource is unrelated to the Storage package and is a collection of (Managed) Disks. Since there's a number of Disk related resources (and will be more with #14268) - this pulls the existing resource and this new resource out into it's own package.

There's a follow up here to move Managed Disks into this package too, but that's outside the scope of this PR.

Dependent on #14673 and #14672

### CODE REVIEW

**Comment by lonegunmanb** [comment]
Hello @tombuildsstuff , should I continue my work #14268 based on this pr once it has been merged? It's your call, awaiting for your instruction.

**Comment by tombuildsstuff** [comment]
Sku needs to be ForceNew in both of these:

```
------- Stdout: -------
=== RUN   TestAccDiskPool_upgradeSku
=== PAUSE TestAccDiskPool_upgradeSku
=== CONT  TestAccDiskPool_upgradeSku
=== CONT  TestAccDiskPool_upgradeSku
testcase.go:110: Step 3/4 error: After applying this test step, the plan was not empty.
stdout:
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
~ update in-place
Terraform will perform the following actions:
# azurerm_disk_pool.test will be updated in-place
~ resource "azurerm_disk_pool" "test" {
id                  = "/subscriptions/*******/resourceGroups/acctestRG-diskspool-211223144533983359/providers/Microsoft.StoragePool/diskPools/acctest-diskspool-w1thg"
name                = "acctest-diskspool-w1thg"
~ sku_name            = "Standard_S1" -> "Premium_P1"
tags                = {}
# (4 unchanged attributes hidden)
}
Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccDiskPool_upgradeSku (1086.98s)
```

**Review by katbyte — APPROVED** [review]
Thanks @tombuildsstuff - LGTM 🏗️

**Pull request merged**
